cmake/Dependencies.cmake

```diff
@@ -179,32 +179,27 @@ if(BUILD_BENCHMARK)
 endif()
 
 # Find or download/install rocm-cmake project
-find_package(ROCM QUIET CONFIG PATHS /opt/rocm)
+find_package(ROCM 0.6 QUIET CONFIG PATHS /opt/rocm)
 if(NOT ROCM_FOUND)
-  set(rocm_cmake_tag "master" CACHE STRING "rocm-cmake tag to download")
-  file(
-    DOWNLOAD https://github.com/RadeonOpenCompute/rocm-cmake/archive/${rocm_cmake_tag}.zip
-    ${CMAKE_CURRENT_BINARY_DIR}/rocm-cmake-${rocm_cmake_tag}.zip
-    STATUS rocm_cmake_download_status LOG rocm_cmake_download_log
-  )
-  list(GET rocm_cmake_download_status 0 rocm_cmake_download_error_code)
-  if(rocm_cmake_download_error_code)
-    message(FATAL_ERROR "Error: downloading "
-      "https://github.com/RadeonOpenCompute/rocm-cmake/archive/${rocm_cmake_tag}.zip failed "
-      "error_code: ${rocm_cmake_download_error_code} "
-      "log: ${rocm_cmake_download_log} "
-    )
-  endif()
-
-  execute_process(
-    COMMAND ${CMAKE_COMMAND} -E tar xzf ${CMAKE_CURRENT_BINARY_DIR}/rocm-cmake-${rocm_cmake_tag}.zip
-    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-    RESULT_VARIABLE rocm_cmake_unpack_error_code
-  )
-  if(rocm_cmake_unpack_error_code)
-    message(FATAL_ERROR "Error: unpacking  ${CMAKE_CURRENT_BINARY_DIR}/rocm-cmake-${rocm_cmake_tag}.zip failed")
+  message(STATUS "rocm-cmake not found. Downloading and building rocm-cmake.")
+  if(CMAKE_CONFIGURATION_TYPES)
+    message(FATAL_ERROR "DownloadProject.cmake doesn't support multi-configuration generators.")
   endif()
```

Comment:
forcing a download is not correct. We should discuss.

Reply:
This is the same as the previous behaviour, it checks to see if it is installed and downloads if not. Only the manner in which the download is performed has changed, in order to build the ROCMConfigVersion.cmake file from the download. This could also have been done with several additional `execute_process` calls.